target/linux/ipq40xx/image/generic.mk

```diff
@@ -1119,14 +1119,14 @@ define Device/tel_x1pro
 	DEVICE_VENDOR := Telco
 	DEVICE_MODEL := X1 Pro
 	SOC := qcom-ipq4019
-	KERNEL_SIZE := 4096k
-	IMAGE_SIZE := 31232k
+	KERNEL_INSTALL := 1
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	FILESYSTEMS := squashfs
```

Comment:
Is this really needed? maybe @robimarko can help?

Reply:
Considering this thing uses SPI-NOR, all of these changes are not needed

Reply:
Without this change I get an error that the image is too big. Can you explain why?

```shell
    WARNING: Image file tel_x1pro-uImage.itb is too big: 4799344 > 4194304
    ERROR: target/linux failed to build.
 ```

Reply:
So you just basically hid that issue by removing the size limit, but this is not the solution.

I think the original limits make no sense anymore, since a single firmware partition with `compatible = "denx,fit";` is used so its split dynamically.

KERNEL_SIZE can probably be safely removed if U-Boot does not have a hardcoded 4MiB limit

Reply:
thanks that makes sense and it is working